02_knight_and_princess/novikov_dmitry/Graph.hpp

```diff
@@ -7,16 +8,23 @@
 #include <unordered_map>
 #include <vector>
 
-constexpr int VERTICES_NUMBER = 14;
+constexpr int DEFAULT_DEPTH = 0;
 
 using VertexId = int;
 using EdgeId = int;
+using Depth = int;
+using EdgeColor = std::string;
 
 class Vertex {
  public:
-  explicit Vertex(const VertexId& new_vertex_id) : id_(new_vertex_id) {}
+  Vertex(const VertexId& new_vertex_id, const Depth& new_vertex_depth)
+      : id_(new_vertex_id), depth_(new_vertex_depth) {}
 
   void add_edge_id(const EdgeId& new_edge_id) {
+    if (std::find(edge_ids_.begin(), edge_ids_.end(), new_edge_id) !=
+        edge_ids_.end()) {
+      return;
+    }

```

Comment:
У нас нет инпута от юзера или из сети, мы только хотим проверить самих себя на адекватность и на баги в коде, для этого существуют `assert`, они обычно работают только в `debug` билдах, и отключаются в `production` коде. Кроме того, логикеу валидации можно вынести в удобный метод интерфейса:
```cpp
bool has_edge_id(const EdgeId& id) const;
```
Тогда это будет выглядеть как-то так:
```suggestion
    assert(!has_edge_id(new_edge_id) && "Edge id already exists")
```